src/eval.c

```diff
@@ -403,6 +403,38 @@ uint64_t evalGetCommandFlags(client *c, uint64_t cmd_flags) {
     return scriptFlagsToCmdFlags(cmd_flags, script_flags);
 }
 
+/* Extracts the shebang flags for both a sha referenced script and a script with
+ * its full body
+ *
+ * Adding to enable modules to make script flag determinations before they
+ * RM_Call() an eval
+ */
+int getScriptFlags(sds sha, sds body, uint64_t *flags, sds *err) {
+    if (sha != NULL) {
+        if (body != NULL) {
```

Comment:
btw I just realized, you can have two functions, one for sha, one for body, so you can avoid these null checks I guess. Just fyi.

Reply:
yes, I guess it's a better approach